File: 1/1-2.js
function calculateTotalModuleFuel(remainingMass, totalFuel=0){
    const fuel = Math.floor(Number(remainingMass) / 3) - 2;
    return fuel <= 0 ? totalFuel : calculateTotalModuleFuel(fuel, totalFuel + fuel)
}

console.log(
    require('fs').readFileSync('masses.txt')
        .toString().split("\n").map(Number)
        .reduce((a, b) => a + calculateTotalModuleFuel(b), 0)
);

File: 1/1-2.py
def calculate_module_fuel(remaining_mass, total_fuel=0):
    fuel = (remaining_mass // 3) - 2
    return total_fuel if fuel <= 0 else calculate_module_fuel(fuel, total_fuel + fuel)


print(list(map(
    lambda x: calculate_module_fuel(int(x.replace('\n', ''))),
    open("masses.txt", "r").readlines()))
)
